realm/src/main/java/io/realm/RealmQuery.java

```diff
@@ -1243,63 +1487,298 @@ public long count() {
     }
 
     /**
+     * Similar to {@link #findAllSorted(String[], boolean[])} but runs asynchronously
+     * from a worker thread
+     *
+     * @return immediately an empty {@link RealmResults}. Users need to register a listener
+     * io.realm.RealmResults#addChangeListener(RealmChangeListener) to be notified
+     * of the query completion.
+     * @throws java.lang.RuntimeException Any other error
+     * @see io.realm.RealmResults
+     */
+    public RealmResults<E> findAllSortedAsync(String fieldNames[], final boolean [] sortAscendings) {
+        if (fieldNames == null) {
```

Comment:
Same validation is done by `findAllSorted()` - maybe a common validation method?